operations/alb-access-logs-anonymizer.yaml

```diff
@@ -172,55 +172,51 @@ Resources:
             if ('versionId' in record.s3.object) {
               params.VersionId = record.s3.object.versionId;
             }
-            return new Promise((resolve, reject) => {
-              const body = stream.pipeline(
-                s3.getObject(params).createReadStream(),
+
+            try {
+              const getObjectResponse = await s3.send(new GetObjectCommand(params));
+
+              const writePipeline = pipeline(
+                getObjectResponse.Body,
                 zlib.createGunzip(),
-                new stream.Transform({
+                new Transform({
                   transform
                 }),
                 zlib.createGzip(),
                 () => {}
               );
-              s3.upload({
-                Bucket: record.s3.bucket.name,
-                Key: anonymizedKey,
-                Body: body
-              }, (err) => {
-                if (err) {
-                  if (err) {
-                    if (err.code === 'NoSuchKey') {
-                      console.log('original no longer exist, check for anonymized object.')
-                      s3.headObject({
-                        Bucket: record.s3.bucket.name,
-                        Key: anonymizedKey
-                      }, (err) => {
-                        if (err) {
-                          reject(err);
-                        } else {
-                          // original already processed
-                          resolve();
-                        }
-                      });
-                    } else {
-                      reject(err);
-                    }
-                  }
-                } else {
-                  if (chunk.length > 0) {
-                    reject(new Error('file was not read completly'));
-                  } else {
-                    s3.deleteObject(params, (err) => {
-                      if (err) {
-                        reject(err);
-                      } else {
-                        resolve();
-                      }
-                    });
+
+              await new Upload({
+                client: s3,
+                params: {
+                  Bucket: record.s3.bucket.name,
+                  Key: anonymizedKey,
+                  Body: writePipeline
+                }
+              }).done();
+            } catch(err) {
+              if (err.Code === 'NoSuchKey') {
+                console.log('original object no longer exists, checking for anonymized object')
+                try {
+                  await s3.send(new HeadObjectCommand({
+                    Bucket: record.s3.bucket.name,
+                    Key: anonymizedKey
+                  }));
+                } catch(err) {
+                  if (err.Code === 'NoSuchKey') {
```

Comment:
what about all the other error Codes? Before, an error was always thrown. Now just NoSuchKey errors are thrown?